### PR TITLE
Add presignedURLV4() to SimpleStorageSigner

### PR DESCRIPTION
**Pull request by drewdeponte**
I did this so to produce a usable v4 pre-signed S3 url and expose it so
consumers of this library can use it.

### CODE REVIEW

File: Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift [hunk 1]
```diff
@@ -57,6 +57,19 @@ extension SimpleStorageSigner {
     func path(_ url: URL) -> String {
         return !url.path.isEmpty ? url.path.encode(with: .pathAllowed) ?? "/" : "/"
     }
+
+    public func presignedURLV4(httpMethod: HTTPMethod, url: URL, expiration: Expiration, headers: [String: String]) throws -> URL? {
+        let dates = Dates(Date())
+        var updatedHeaders = headers
+        updatedHeaders["Host"] = url.host ?? region.host
```
Comment (anthonycastelli) [inline Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift]: I prefer to use `self` when referencing variables in the class. Can you add that to here `self.region.host`

File: Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift [hunk 2]
```diff
@@ -57,6 +57,19 @@ extension SimpleStorageSigner {
...
+
+        let (canonRequest, fullURL) = try presignedURLCanonRequest(httpMethod, dates: dates, expiration: expiration, url: url, headers: updatedHeaders)
```
Comment (anthonycastelli) [inline Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift]: Same thing here `self. presignedURLCanonRequest`

File: Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift [hunk 3]
```diff
@@ -57,6 +57,19 @@ extension SimpleStorageSigner {
...
+
+        let stringToSign = try createStringToSign(canonRequest, dates: dates)
+        let signature = try createSignature(stringToSign, timeStampShort: dates.short)
```
Comment (anthonycastelli) [inline Sources/SimpleStorageSigner/SimpleStorageSigner+Signing.swift]: And again for line 68 & 69

**Review by anthonycastelli — CHANGES_REQUESTED** [review]
Looks good!

**Comment by drewdeponte** [comment]
@anthonycastelli I pushed up the requested changes.

**Review by anthonycastelli — APPROVED** [review]

**Comment by anthonycastelli** [comment]
👍 I have to go back at some point and clean up this entire framework, but looks good for now.

**Pull request merged**